ibu-imager/postpivot/postpivot.go

```diff
@@ -0,0 +1,250 @@
+	if err != nil {
+		return fmt.Errorf("failed to get seed info from %s, err: %w", "", err)
+	}
+
+	p.log.Info("Create recert configuration file")
+	if err := recert.CreateRecertConfigFile(clusterInfo, seedClusterInfo, path.Join(p.workingDir, common.CertsDir),
+		p.workingDir); err != nil {
+		return err
+	}
+
+	if err := p.recert(clusterInfo, seedClusterInfo); err != nil {
+		return err
+	}
+
+	client, err := utils.CreateKubeClient(p.scheme, p.kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to create k8s client, err: %w", err)
+	}
+
+	if _, err := p.ops.SystemctlAction("enable", "kubelet", "--now"); err != nil {
+		return err
+	}
+	p.waitForApi(ctx, client)
+	p.approveCsrs(ctx, client)
+
+	if err := p.applyManifests(); err != nil {
+		return err
+	}
+
+	if _, err = p.ops.SystemctlAction("disable", "installation-configuration.service"); err != nil {
+		return fmt.Errorf("failed to disable installation-configuration.service, err: %w", err)
+	}
+
+	p.log.Infof("Removing %s", p.workingDir)
+	return os.RemoveAll(p.workingDir)
+}
+
+func (p *PostPivot) recert(clusterInfo, seedClusterInfo *clusterinfo.ClusterInfo) error {
+	doneFile := path.Join(p.workingDir, recertDone)
+	if _, err := os.Stat(doneFile); err == nil {
+		p.log.Infof("Found %s file, skippin recert", doneFile)
+		return nil
+	}
+
+	if err := p.ops.RunUnauthenticatedEtcdServer(p.authFile, common.EtcdContainerName); err != nil {
+		return fmt.Errorf("failed to run etcd, err: %w", err)
+	}
+
+	defer func() {
+		p.log.Info("Killing the unauthenticated etcd server")
+		if _, err := p.ops.RunInHostNamespace("podman", "kill", common.EtcdContainerName); err != nil {
+			p.log.WithError(err).Errorf("failed to kill %s container.", common.EtcdContainerName)
+		}
+
+		if _, err := p.ops.RunInHostNamespace("podman", "rm", common.EtcdContainerName); err != nil {
+			p.log.WithError(err).Errorf("failed to rm %s container.", common.EtcdContainerName)
+		}
+	}()
+
+	if err := p.additionalCommands(clusterInfo, seedClusterInfo, common.EtcdContainerName); err != nil {
+		return err
+	}
+
+	if err := p.ops.RunRecert(p.recertContainerImage, p.authFile, path.Join(p.workingDir, recert.RecertConfigFile),
+		"-v", fmt.Sprintf("%s:%s", p.workingDir, p.workingDir)); err != nil {
+		return err
+	}
+
+	_, err := os.Create(doneFile)
+	return err
+}
+
+func (p *PostPivot) additionalCommands(clusterInfo, seedClusterInfo *clusterinfo.ClusterInfo, etcdImage string) error {
+	// TODO: remove after https://issues.redhat.com/browse/ETCD-503
+	newEtcdIp := clusterInfo.MasterIP
+	if utils.IsIpv6(newEtcdIp) {
+		newEtcdIp = fmt.Sprintf("[%s]", newEtcdIp)
+	}
+	// TODO: move to etcd client?
+	_, err := p.ops.RunInHostNamespace("podman", "exec", "-it", etcdImage, "bash", "-c",
+		fmt.Sprintf("/usr/bin/etcdctl member list | cut -d',' -f1 | xargs -i etcdctl member update \"{}\" --peer-urls=http://%s:2380", newEtcdIp))
+	if err != nil {
+		return fmt.Errorf("failed to change etcd peer url, err: %w", err)
+	}
```

Comment:
```suggestion
	if _, err := p.ops.RunInHostNamespace("podman", "exec", "-it", etcdImage, "bash", "-c",
		fmt.Sprintf("/usr/bin/etcdctl member list | cut -d',' -f1 | xargs -i etcdctl member update \"{}\" --peer-urls=http://%s:2380", newEtcdIp)); err != nil {
		return fmt.Errorf("failed to change etcd peer url, err: %w", err)
	}
```

Reply:
We should not do this type of things all the time, it is very long command it should not be part of "if"